src/Engine/Graphics/IRender.h

```diff
@@ -369,7 +375,8 @@ class IRender {
     int hd_water_current_frame;
     GraphicsImage *hd_water_tile_anim[7];
     RenderBillboardD3D pBillboardRenderListD3D[1000];
-    unsigned int uNumBillboardsToDraw;
+    unsigned int uNumBillboardsToDraw; // TODO(captainurist): this is not properly cleared if BeginScene3D is not called,
+                                       //                     resulting in dangling textures in pBillboardRenderListD3D.
```

Comment:
@pskelton can you help me with this TODO? What I need is to have this set properly so that there's never dangling textures in pBillboardRenderListD3D

Reply:
Yes, I will sort this. IIRC it doesn't get cleared until you start drawing the next frame - we use last frames billboards for mouse picking so that will need to be addressed too.